src/megaapi_impl.cpp

```diff
@@ -18378,6 +18378,14 @@ void MegaApiImpl::sendPendingRequests()
             const char *name = request->getName();
             if(!name || !(*name) || !parent) { e = API_EARGS; break; }
 
+            // prevent to create a duplicate folder with same name in same path
+            Node *folder = client->childnodebyname(parent, name, false);
+            if (folder && folder->type == FOLDERNODE)
+            {
+                e = API_EEXIST;
+                break;
```

Comment:
Given this is not backward compatible, it will break any existing logic relying on it, won't it?

If this is meant to work like in the webclient, as pointed in the Redmine ticket, please note there we do not fail when the folder exists, we do pretend the operation was successful and we do return the node's handle for the existing folder.